### PR TITLE
Backport unzip method in functor typeclass for scala_2.11

### PR DESCRIPTION
**Pull request by gagandeepkalra**
addresses #3143

### CODE REVIEW

**Comment by gagandeepkalra** [comment]
If this looks good I can start working on other subtask for #3143

**Comment by gagandeepkalra** [comment]
I need some help with this error on running `docs/makeMicrosite`

> [error] (docs / Jekyll / mappings) java.io.IOException: Cannot run program "jekyll" (in directory "/Users/gagandeepkalra/IdeaProjects/cats/docs/target/scala-2.11/resource_managed/main/jekyll"): error=2, No such file or directory`

Seems like it's failing for scala_2.11 branch altogether, I'm not very sure how to solve this.

**Comment by rossabaker** [comment]
Have you installed jekyll?  It's a Ruby utility, and the right way to [install it](https://jekyllrb.com/docs/installation/#guides) depends on your OS.

**Comment by gagandeepkalra** [comment]
@rossabaker thank you, now I have, and `docs/makeMicrosite` runs successfully on both my branch and `scala_2.11`. Now I wonder why is the build failing at all. I tried restarting with a dummy commit (amend) thinking it might just be flaky, but no.

build error,
>[error] /home/travis/.rvm/gems/ruby-2.5.3/gems/octokit-4.14.0/lib/octokit/middleware/follow_redirects.rb:14:in `<module:Middleware>': uninitialized constant Faraday::Error::ClientError (NameError)
[error] Did you mean?  Faraday::ClientError

ends up with.
>[error] (docs / Jekyll / mappings) Could not run jekyll, error: 1

**Comment by rossabaker** [comment]
It suspect it's related to https://github.com/typelevel/cats/pull/3218.  There was some recent bitrot in jekyll that broke several projects using sbt-microsites, and it doesn't look like any fix was backported to this branch.

File: core/src/main/scala/cats/syntax/all.scala [hunk 1]
```diff
@@ -93,3 +94,5 @@ trait AllSyntaxBinCompat4
 trait AllSyntaxBinCompat5 extends ParallelBitraverseSyntax
 
 trait AllSyntaxBinCompat6 extends ParallelUnorderedTraverseSyntax
+
+trait AllSyntaxBinCompat7 extends FunctorSyntaxBinCompat0
```
Comment (rossabaker) [inline core/src/main/scala/cats/syntax/all.scala]: I can't remember why some syntax traits are package private and some aren't.  I can't think of a reason this couldn't and shouldn't be, but I also was not very active in developing this idiom.

Comment (travisbrown) [reply core/src/main/scala/cats/syntax/all.scala]: @rossabaker As far as I know there's not a good answer to this question—like a lot of stuff in Cats it's just the result of a lot of uncoordinated decisions. I opened a related issue [here](https://github.com/typelevel/cats/issues/3008) a few months ago.

Comment (rossabaker) [reply core/src/main/scala/cats/syntax/all.scala]: @gagandeepkalra has been through a lot of incidental distractions already. I'm good with punting on it for the purposes of this ticket.

**Comment by gagandeepkalra** [comment]
Thank you 😊

File: core/src/main/scala/cats/syntax/all.scala (hunk 1 above)
Comment (gagandeepkalra) [reply core/src/main/scala/cats/syntax/all.scala]: thank you

**Pull request merged**